packages/metrics/src/testing/MetricsAsserter.ts

```diff
@@ -36,30 +36,24 @@ class AssertionHelper {
   }
 }
 
-type TestingConfig = Pick<Config, 'SERVICE_NAME'>;
-/** Doing this once in module scope is fine since only 1 service will be running tests, there is no realistic risk of collision. */
-const DEFAULT_SERVICE_NAME = `test-${Math.random().toString(36).slice(2)}`;
-
-export class MetricsAsserter<M extends Metrics> {
-  constructor(private readonly MetricsClient: (config: Config) => M) {}
+export class MetricsAsserter<Metrics extends BaseMetrics, Config extends Record<never, never>> {
+  constructor(private readonly MetricsClient: (config: Config & BaseConfig) => Metrics) {}
```

Comment:
Should this be `Partial<Config>` to avoid the `unknown` type conversion below? Since `assertMetric` allows a partial config, the typing isn't correct when invoking `MetricsClient` and a config that is assumed to be there may not be.

Reply:
tldr: no

longer explanation:

first, some context - this is how this class is used to begin writing tests:

```js
import MetricsClient from '@/client';
import * as VFMetrics from '@/index';

const metricsAsserter = new VFMetrics.Testing.MetricsAsserter(MetricsClient);
```

`MetricsClient` is a function that users define as something like `(config: MyConfig) => MyMetrics`

if we change the constructor ro require a `(config: Partial<MyConfig>) => MyMetrics` typescript is going to lose it's mind and say "hey wait a minute, `MetricsClient` isn't compatible with `Partial<MyConfig>`, you need to pass in `Config`".

so then users are going to need to write this:

```ts
const metricsAsserter = new VFMetrics.Testing.MetricsAsserter(MetricsClient as any);
```

a type assertion is necessary regardless*, we can either do it once in the library or every time you call `new MetricsAsserter`. i think doing it here in the library is cleaner.

*: the absolute safest way to deal with this is to not allow a `Partial` anywhere, users must pass in a fully defined `Config` object. this will result in very hideous code, since you usually only want to tweak a singular env var, not the entire config